src/python/WMCore/WorkQueue/DataStructs/WorkQueueElement.py

```diff
@@ -243,33 +240,20 @@ def passesSiteRestriction(self, site):
 
         return True
 
-    def intersectionWithEmptySet(self, a, b):
-        """
-        interaction of 2 sets but if one of the set is empty returns union
-        """
-        if len(a) == 0 or len(b) == 0:
-            return a | b
-        else:
-            return a & b
-
     def possibleSites(self):
```

Comment:
@ticoann as we just discussed, we are/were going to make an empty SiteWhitelist as "consider ALL sites", as it's right now AFAICT.
After staring at the code for 10min :-) I noticed there is one use case that we cannot really distinguish:
 i) a MonteCarlo element would NOT have Inputs, ParentData nor PileupData, thus only the SiteWhitelist that would be empty. Value returned by this function is `[]` --> meaning run everywhere.
 ii) any other element with either Input/Parent/Pileup should follow the intersection of data location. Thus it could return `[]` --> meaning no possible location.

So we'll have this inconsistency :( Maybe we should just make SiteWhitelist mandatory and non empty during assignment. Checking with Jean-Roch to see if they have ever used it....